src/almanac.c

```diff
@@ -36,8 +37,42 @@
  * \param vel  The satellite velocity in ECEF coordinates is returned in this
  *             vector. Ignored if NULL.
  */
-void calc_sat_state_almanac(const almanac_t* alm, double t, s16 week,
-                            double pos[3], double vel[3])
+void sbas_calc_sat_state_almanac(const sbas_almanac_t* alm, double t,
+                                 double pos[3], double vel[3])
+{
+  u8 nweeks = t / 86400;
+  double tod = t - 86400 * nweeks;
+  double dt = tod - alm->t0;
+
```

Comment:
explain these magic numbers (seconds in day and seconds in day/2)  explain this logic for determining the dt.